### ISSUE TITLE
PERF401 false negative

### ISSUE THREAD

**Issue by victorsamun** [astral-sh/ruff#21648]
### Summary

When using unpacking in for-loop variable rule PERF401 does not work.

Wrong behaviour:
```bash
$ cat 1.py && ruff check --select PERF401 1.py
xs = []
for x, y in f():
    xs.append(x + y)
All checks passed!
```

Correct behaviour:
```bash
$ cat 2.py && ruff check --select PERF401 2.py
xs = []
for xy in f():
    xs.append(xy[0] + xy[1])
PERF401 Use a list comprehension to create a transformed list
 --> 2.py:3:5
  |
1 | xs = []
2 | for xy in f():
3 |     xs.append(xy[0] + xy[1])
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |
help: Replace for loop with list comprehension

Found 1 error.
```

### Version

0.14.6
